src/query/stream/streams.cpp

```diff
@@ -724,15 +725,27 @@ TransformationResult Streams::Check(const std::string &stream_name, std::optiona
           auto accessor = interpreter_context->db->Access();
           CallCustomTransformation(transformation_name, messages, result, accessor, *memory_resource, stream_name);
 
-          for (auto &row : result.rows) {
-            auto [query, parameters] = ExtractTransformationResult(row.values, transformation_name, stream_name);
-            std::vector<TypedValue> result_row;
-            result_row.reserve(kExpectedTransformationResultSize);
-            result_row.push_back(std::move(query));
-            result_row.push_back(std::move(parameters));
+          auto result_row = std::vector<TypedValue>();
+          result_row.reserve(kCheckStreamResultSize);
 
-            test_result.push_back(std::move(result_row));
-          }
+          auto queries_and_parameters = std::vector<TypedValue>(result.rows.size());
+          std::transform(
+              result.rows.cbegin(), result.rows.cend(), queries_and_parameters.begin(), [&](const auto &row) {
+                auto [query, parameters] = ExtractTransformationResult(row.values, transformation_name, stream_name);
+
+                return std::map<std::string, TypedValue>{{"query", std::move(query)},
+                                                         {"parameters", std::move(parameters)}};
+              });
+          result_row.emplace_back(std::move(queries_and_parameters));
+
+          auto messages_list = std::vector<TypedValue>(messages.size());
+          std::transform(messages.cbegin(), messages.cend(), messages_list.begin(), [](const auto &message) {
+            return std::string_view(message.Payload().data(), message.Payload().size());
```

Comment:
`return std::string(....) `;

Reply:
Why `std::string`? It copies the data from payload, then when the TypedValue is constructed, it will also copy it into a `std::pmr::string`. With the `string_view` you can avoid the first copy.

Reply:
Conclusion is we'll go with `std::string_view`.
From @antaljanosbenjamin -> `TypedValue` doesn't support construction with not null terminated strings, so we have to construct the a `pmr::string` , that will be moved into the `TypedValue` that sits inside the vector. However if somebody changes that vector to pmr::vector, that might still result in copying `pmr::string`. With `std::string_view`, we copy a small object (if compiler doesn't optimize the whole stuff away, as `std::transform`, `TypedValue` and also `pmr::string` is a template type implemented in header files), but get rid of the complexity of allocators. In my view this is not that a big issue (we are not on the hot path), so I think the gained safety worth more than the performance we could gain with `pmr::string`.